dashboard/src/pages/macros.tsx

```diff
@@ -90,6 +100,44 @@ const Macros = () => {
     );
   };
 
+  const openMacroModal = async (row: TableRow) => {
+    if (!selectedInstance) {
+      toast.error('Error running macro: No instance selected');
+      return;
+    }
+    const macroData = await getMacroConfig(selectedInstance.uuid,  row.name as string);
```

Comment:
Maybe display the returned error message as well? This could tell the user why the edit config tab is popping out when clicking on run macro. Try to use returned error data instead of the message, as I might remove the message return later in the backend